agent/agentcontainers/containers_internal_test.go

```diff
@@ -93,12 +101,88 @@ func TestDockerCLIContainerLister(t *testing.T) {
 			if assert.Len(t, foundContainer.Volumes, 1) {
 				assert.Equal(t, testTempDir, foundContainer.Volumes[testTempDir])
 			}
+			// Test command execution
+			cmd := wex.CommandContext(ctx, "cat", "/etc/hostname")
+			out, err := cmd.CombinedOutput()
+			if !assert.NoError(t, err) {
+				t.Logf("Container %q exited with error: %v", ct.Container.ID, err)
+				t.Logf("Output:\n%s", string(out))
+				t.FailNow()
+			}
+			require.Equal(t, ct.Container.Config.Hostname, strings.TrimSpace(string(out)))
+			// Test command execution with PTY
+			ptyCmd, ptyPs, err := pty.Start(wexpty.PTYCommandContext(ctx, "/bin/sh"))
+			require.NoError(t, err, "failed to start pty command")
+			t.Cleanup(func() {
+				_ = ptyPs.Kill()
+				_ = ptyCmd.Close()
+			})
+			_, _ = ptyCmd.InputWriter().Write([]byte("hostname\r\n"))
+			var ptyOut []byte
+			_, err = ptyCmd.OutputReader().Read(ptyOut)
+			require.NoError(t, err, "failed to read pty output")
+			require.Equal(t, ct.Container.Config.Hostname, strings.TrimSpace(string(out)))
```

Comment:
review: open to suggestions on improving this, I wasn't sure how exactly to wire up `ptytest` here.